radixdlt/src/integration/java/com/radixdlt/integration/distributed/simulation/application/EpochsNodeSelector.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * (C) Copyright 2020 Radix DLT Ltd
+ *
+ * Radix DLT Ltd licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.  See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+package com.radixdlt.integration.distributed.simulation.application;
+
+import com.google.common.collect.ImmutableList;
+import com.radixdlt.consensus.bft.BFTNode;
+import com.radixdlt.consensus.bft.BFTValidator;
+import com.radixdlt.integration.distributed.simulation.network.SimulationNodes.RunningNetwork;
+import io.reactivex.rxjava3.core.Single;
+import java.util.Random;
+
+/**
+ * Selects random nodes from changing validator sets given by epochs
+ */
+public class EpochsNodeSelector implements NodeSelector {
+	private final Random random = new Random();
+
+	@Override
+	public Single<BFTNode> nextNode(RunningNetwork network) {
+		return network.latestEpochChanges()
+			.map(e -> {
+				ImmutableList<BFTValidator> validators = e.getBFTConfiguration()
+					.getValidatorSet().getValidators().asList();
+				int validatorSetSize = validators.size();
```

Comment:
Long call chain suggests that we're looking too deep into class internals. Probably we should have method size() in BFTValidatorSet. There are several locations where this call sequence is repeated, so we could reduce boilerplate and improve readability.

Reply:
Agreed though I'd rather do this change in another PR